File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsJava/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicsjava/ui/AppViewModel.java
package pl.edu.uwr.pum.dagger_hilt_room_basicsjava.ui;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.Student;
import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.db.AppDatabase;
import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.repository.AppRepository;

@HiltViewModel
public class AppViewModel extends AndroidViewModel {

    private final AppRepository repository;

    @Inject
    public AppViewModel(Application app, AppRepository repository) {
        super(app);
        this.repository = repository;
    }

    LiveData<List<Student>> readAllData(){
        return repository.readAllData();
    }

    void insert(Student student){
        AppDatabase.databaseWriteExecutor.execute(() -> repository.insert(student));
    }
}

File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicskotlin/ui/AppViewModel.kt
package pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.ui

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.Student
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.domain.repository.AppRepository
import javax.inject.Inject

@HiltViewModel
class AppViewModel @Inject constructor(
    app: Application,
    private  val repository: AppRepository
) : AndroidViewModel(app) {

    val readAllData: LiveData<List<Student>> = repository.readAllData

    fun insert(student: Student) = viewModelScope.launch {
        repository.insert(student)
    }
}